flow/project.py

```diff
@@ -330,13 +468,18 @@ def __init__(self, id, name, job, cmd, directives=None):
         self.directives._keys_set_by_user = keys_set_by_user
 
     def __str__(self):
-        return "{}({})".format(self.name, self.job)
+        if len(self.jobs) == 1:
+            return "{}({})".format(self.name, str(self.jobs[0]))
+        else:
+            return "{}({}...{}...{})".format(
+                self.name, str(self.jobs[0])[0:8], len(self.jobs), str(self.jobs[-1])[0:8]
+            )
 
     def __repr__(self):
-        return "{type}(name='{name}', job='{job}', cmd={cmd}, directives={directives})".format(
+        return "{type}(name='{name}', job(s)='{job}', cmd={cmd}, directives={directives})".format(
                    type=type(self).__name__,
                    name=self.name,
-                   job=str(self.job),
+                   job=" ".join(map(str, self.jobs)),
```

Comment:
```suggestion
                   jobs=" ".join(map(str, self.jobs)),
```
These changes are more in line with what `repr` is supposed to do versus `str`.